### ISSUE TITLE
Error or misbehavior if two part of the application use a struct with the same name

### ISSUE THREAD

**Issue by npwoods** [slint-ui/slint#9358]
Edit: If two slint file have a struct or enum with the same name, we get complation error or bugs in Rust and C++.
This was also reported in #6880 and #10723 with more testcases.

### Original Bug Description

This is a weird weird issue that seems to result from using `vivi_ui` and `material-components` in conjunction.

In the attached example, there is a `NavigationDrawer` on the left and a `StandardTableView` on the right, but the `StandardTableView` is not using all of the space available to it.

One thing that will cause the `StandardTableView` to start using all available space is if the `MenuBar` is removed.  This is bizarre.

A possible clue is that for some reason, I have to put a dummy `MagicListView` and if I do not, the Slint compiler throws errors.

This sample is as minimal as I could make it.  For some reason it is necessary to have the other `paths.slint` file present or else the problem goes away.

Code attached:
[bug_demo.zip](https://github.com/user-attachments/files/22199825/bug_demo.zip)

The following screenshot shows the problem; there is empty space at the bottom that I would expect to be used:
<img width="1419" height="1011" alt="Image" src="https://github.com/user-attachments/assets/02c01468-7b67-47c4-a3da-869dfbccc128" />

This screenshot shows what happens when the menu bar is removed; the space is properly used
<img width="951" height="1261" alt="Image" src="https://github.com/user-attachments/assets/44114342-197e-4c79-9768-aed2428a2814" />

Inside the sample is a dummy `MagicListView` looking like this:
```
        MagicListView {
            width: 0px;
            height: 0px;
            visible: false;
        }
```
If this is removed, we get the following compilation errors:  [build_log_when_dummy_magic_list_view_is_removed.txt](https://github.com/user-attachments/files/22199720/build_log_when_dummy_magic_list_view_is_removed.txt)

### Environment Details

- Slint Version: 1.13
- Platform/OS: Windows 11
- Programming Language: Rust
- Backend/Renderer: Default

**Comment by ogoffart** [comment]
The error from the log is 
```
error[E0609]: no field `foreground` on type `TextStyle`
     --> C:\Temp\bug-demo\target\debug\build\bug-demo-25f2b995c4b13362\out\main.rs:16892:55
      |
16892 |                     ) . apply_pin (_self) . get ()) . r#foreground) as _ }
      |                                                       ^^^^^^^^^^^^ unknown field
      |
      = note: available fields are: `font_size`, `font_weight`
```

... And more error about `TextStyle`

I believe that there is a `TextStyle` struct define in both libraries and they conflict which eachother.

We should probably mangle these and only the one that ends up re-exported in the public API should get the original name.

I have no idea why MagicListView changes anything.

Similar to https://github.com/slint-ui/slint/issues/9094
